src/t_string.c

```diff
@@ -89,17 +97,21 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
     setkey_flags |= ((flags & OBJ_KEEPTTL) || expire) ? SETKEY_KEEPTTL : 0;
     setkey_flags |= found ? SETKEY_ALREADY_EXIST : SETKEY_DOESNT_EXIST;
 
-    setKeyWithDictEntry(c,c->db,key,val,setkey_flags,de);
+    setKeyByLink(c, c->db, key, val, setkey_flags, &link);
+    if (expire) *val = setExpireByLink(c,c->db, key->ptr, milliseconds, link);
+
+    /*  refcnt 1->2. referenced by the key in the db and by the caller. */
+    incrRefCount(*val);
```

Comment:
it's hard to understand, `val` was changed in setKeyByLink() and c->argv was changed too, to we have to incr the refcount for c->argv for argv release, maybe we can improve the comment.

Reply:
I was wondering if we could modify `setExpireByLink(... , robj **valref, …)` to `setExpireByLink(... , robj *val, …, robj **new_kvobj)`, and then we don't modify val, but the caller releases it, because valref has actually changed from robj to kvobj at this point, which also avoids ambiguity.
This also avoids the problem of the caller needing to incr refcount.

Reply:
If so, we can improve the comment, otherwise it is not easy to understand.

Reply:
Done. Please review the new comment.